### ISSUE TITLE
system.core maintenance_theme does not affect site in maintenance mode

### ISSUE THREAD

**Issue by rszrama** [backdrop/backdrop-issues#1616]
I tried for a couple hours this afternoon to make that sucker work to support a theme I was working on, and I couldn't get it to. It appears that by the time the maintenance page is rendered through backdrop_deliver_html_page(), the global $theme has already been initialized... meaning when _backdrop_maintenance_theme() gets called, it bails right out without ever check the relevant system.core setting. No matter what I tried, and I tried just about every global variable and hook there is (including hook_custom_theme()), I couldn't get in there early enough to affect the theme. I finally resorted to a hook_init() that simply re-initializes the theme to the one I actually wanted.

It appears, unless I'm mistaken, that the variable is actually only ever coming into play for install.php and update.php. I'm more than happy to be proven wrong, otherwise we should probably fix core so it alters the maintenance page as well. : )

**Comment by gifad** [comment]
Also note, setting `$settings['maintenance_theme']` in `/settings.php` exhibits the same behavior…

**Comment by serundeputy** [comment]
It is true `$theme` is already set so the `_backdrop_maintenance_theme()` function bails right away.

```php
// If $theme is already set, assume the others are set too, and do nothing.
  if (isset($theme)) {
    return;
  }
```

I propose we simply remove that snippet; it will respect the `system.core.maintenance_theme` variable.

I can't see any harm in removing that check.  Can't be much of a performance hit can it?  Plus we are in maintenance mode.

Objections? other options?

**Comment by rszrama** [comment]
I'm not sure we want the rest of that function to execute regardless. It would result in an additional theme initialization, for example, which is the hack I have in place now. Additionally, I don't think we'd need this code to execute:

```php
  // These are usually added from system_init() -except maintenance.css.
  // When the database is inactive it's not called so we add it here.
  $path = backdrop_get_path('module', 'system');
  backdrop_add_css('core/misc/normalize.css', array('group' => CSS_SYSTEM, 'every_page' => TRUE));
  backdrop_add_css($path . '/css/system.css');
  backdrop_add_css($path . '/css/system.admin.css');
  backdrop_add_css($path . '/css/system.theme.css');
  backdrop_add_css($path . '/css/system.maintenance.css');
```

**Comment by rszrama** [comment]
(I'll keep it simple, but even better... https://github.com/codewombat/backshop ; )

**Comment by klonos** [comment]
:open_mouth: :tada: :clap:

**Comment by docwilmot** [comment]
Great to hear, and see.

**Comment by quicksketch** [comment]
Huh. You're not crazy.

I can't see how this is supposed to work. The code definitely isn't set up for it. Here's what I did:

- Manually edited system.core.json to set `"maintenance_theme": "seven",`.
- Turned on maintenance mode.
- Log out.
- Site is still in Bartik.
- Edited settings.php and set `$settings['maintenance_theme'] = 'seven';`
- Site is still in Bartik.

But then I tried this all in Drupal 7. *It doesn't work there either.* Looks like maintenance mode theme does not currently work in either Backdrop or D7. As noted in the original report, it *does* show up on update.php however. So right now, the default theme is always used as the maintenance theme.
